src/test/java/org/jenkinsci/plugins/credentialsbinding/impl/BindingStepTest.java

```diff
@@ -240,6 +258,53 @@
         });
     }
 
+    @Issue("JENKINS-30326")
+    @Test
+    public void testGlobalBindingWithAuthorization() {
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                // configure security
+                story.j.jenkins.setSecurityRealm(story.j.createDummySecurityRealm());
+                story.j.jenkins.setAuthorizationStrategy(new FullControlOnceLoggedInAuthorizationStrategy());
+                // create the user.
+                story.j.jenkins.getUser("dummy");
+                
+                // enable the run as user strategy for the AuthorizeProject plugin
+                Map<String, Boolean> strategies = new HashMap<String, Boolean>();
+                strategies.put(story.j.jenkins.getDescriptor(SpecificUsersAuthorizationStrategy.class).getId(), true);
+                QueueItemAuthenticatorConfiguration.get().getAuthenticators().add(new ProjectQueueItemAuthenticator(strategies));
```

Comment:
BTW it is common to directly implement a particular `QueueItemAuthenticator` as a test extension, so that you do not need to actually depend on `authorize-project`. But this is fine too, of course—just makes it a bit more like an integration test and a bit less like a unit test.